### PR TITLE
Rust quickstart example and trait bounds + hacks to make it work.

### PR DESCRIPTION
**Pull request by saolsen**

### CODE REVIEW

File: languages/rust/oso/src/host/class.rs [hunk 1]
```diff
@@ -215,3 +215,5 @@ pub struct Instance {
     pub attributes: Arc<InstanceMethods>,
     pub methods: Arc<InstanceMethods>,
 }
+
+unsafe impl Send for Instance {}
```
Comment (samscott89) [inline languages/rust/oso/src/host/class.rs:219]:
```suggestion
// @TODO: This is very unsafe.
// Temporary workaround. We need to differentiate between instances which 
// _do_ need to be `Send` (e.g. registered as constants on the base `Oso` objects)
// and instances which don't need to be Send (e.g. created/accessed on a single thread for
// just one query).
unsafe impl Send for Instance {}
```

File: languages/rust/oso/src/oso.rs [hunk 2]
```diff
@@ -13,7 +12,7 @@ use crate::ToPolar;
 
 #[derive(Clone)]
 pub struct Oso {
-    inner: Rc<polar_core::polar::Polar>,
+    inner: Arc<polar_core::polar::Polar>,
     host: Arc<Mutex<Host>>,
```
Comment (samscott89) [inline languages/rust/oso/src/oso.rs:16]: Follow up: I don't think we need this to be shared between all queries - typically we clone the host and pass to the query. Would remove most of the contention in exchange for a more expensive clone up front.

Comment (dhatch) [reply languages/rust/oso/src/oso.rs:16]: Have this noted down in my ticket.

**Review by dhatch — APPROVED** [review]
This looks good, minus the Send hack.  I will add that to the list on the catch all hard stuff ticket (Mutexes + Semaphores).  

I haven't checked that it is actually sufficient to write the docs, or compared it against any of the other examples.

**Pull request merged**